src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -3454,8 +3454,8 @@ private void recordRelocation(void* location, void* target, ushort fRelocType, u
             Debug.Assert(locationBlock != BlockType.Unknown, "BlockType.Unknown not expected");
 
             // TODO: Arbitrary relocs
-            if (locationBlock != BlockType.Code)
-                throw new NotImplementedException("Arbitrary relocs");
+            // if (locationBlock != BlockType.Code)
```

Comment:
Yes, that code is for `targetBlock`. This check is for `locationBlock `. The code below does not handle `locationBlock != BlockType.Code`. You will get a pretty spectacular crash once you start running the method where this problem happened - the reloc will be applied over a random instruction.